ops/terraform/env/mgmt/ecr.tf

```
@@ -39,3 +44,93 @@ resource "aws_ecr_repository" "bfd" {
     scan_on_push = true
   }
 }
+
+resource "aws_ecr_registry_policy" "for_bfd" {
+
+  policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+      #Deny request if the Principal is not an Assumed Role with an IAM role in the account
+      {
+        Effect = "Deny",
+        Principal = {
+          "AWS" : "*"
+        },
+        Action = ["ecr:*"],
+        Condition = {
+          "StringNotLike" = {
+            "aws:PrincipalType" = "AssumedRole"
+          },
+          "StringNotEquals" : {
+            "aws:SourceAccount" : "${local.account_id}"
+          }
+        }
+      },
+
+      #Deny destructive actions on the repository if user does not have the administrative role to perform the action
+      {
+        Effect = "Deny",
+        Principal = {
+          "AWS" : "*"
+        },
+        Action = [
+          "ecr:BatchDeleteImage",
+          "ecr:DeleteRepository",
+          "ecr:DeleteRepositoryPolicy",
+          "ecr:PutImageTagMutability",
+          "ecr:DeleteLifecyclePolicy"
+        ],
+        Condition = {
+          "ArnNotEquals" = {
+            "aws:PrincipalArn" = local.admin_roles_arn
+          }
+        }
+      },
+
+      #untagged images are not allow
+      {
+        Effect = "Deny",
+        Action = "ecr:PutImage",
+        Condition = {
+          "StringLike" = {
+            "ecr:ImageTag" : [""]
+          }
+        }
+      }
+    ]
+  })
+}
+
+resource "aws_ecr_lifecycle_policy" "for_bfd_repositories" {
```

```suggestion
resource "aws_ecr_lifecycle_policy" "this" {
```

See comment above for context around this suggestion.